### PR TITLE
Editing and deleting goal data in the database

### PR DESCRIPTION
**Pull request by shiryz**
Editing and deleting goal functionality on the server side

Amended to build on changes in #260 

^^ description by Emily!

### CODE REVIEW

File: server/database/handle-goal-data.js [hunk 1]
```diff
@@ -42,7 +65,14 @@ module.exports = (goal, user_id, callback) => {
...
       // adapt this when we have editing and deleting functionality
-      updateGoal(getGoalResult.rows[0], goal, callback);
+      const oldGoal = getGoalResult.rows[0];
+      if (goal.deleted) {
+        deleteGoal(goal, callback);
+      } else if (goal.edited) {
+        editGoal(goal, callback);
+      } else {
+        updateGoal(oldGoal, goal, callback);
+      }
```
Comment (emilyb7) [inline server/database/handle-goal-data.js]: I guess there's also the case that the user might have edited the goal AND added a rating. In which case we should probs also run updateGoal after editing...

Comment (shiryz) [inline server/database/handle-goal-data.js]: if I call the update with no ratings changed, will that give an error?

Comment (emilyb7) [inline server/database/handle-goal-data.js]: nope, it'll be fine!

**Comment by emilyb7** [comment]
let's wait for #260 before we fix this one

File: server/controller/receive-goal-data.js [hunk 2]
```diff
@@ -1,23 +1,43 @@
...
+    1. get details of goal from DB
+    2. if goal edited on client, change name
+    3. get existing ratings for goal from DB
+    4. add any missing ratings
+    5. if goal doesn't exist yet, add goal to DB
+    6. if newly added goal already has new ratings, add those ratings
+    
+*/
+
+const handleDeleted = (goal, cb) => {
```
Comment (des-des) [inline server/controller/receive-goal-data.js]: Urm, this is the same as 
```js
const handleDeleted = deleteGoalDb
```

**Pull request merged**